usecases/objects/references_delete.go

```diff
@@ -73,32 +94,14 @@ func (m *Manager) DeleteObjectReference(ctx context.Context, principal *models.P
 			input.Reference.Beacon = toBeacon
 		}
 	}
-	if err := m.authorizer.Authorize(principal, authorization.READ, authorization.CollectionsMetadata(input.Reference.Class.String())...); err != nil {
-		return &Error{err.Error(), StatusForbidden, err}
-	}
-
-	res, err := m.getObjectFromRepo(ctx, input.Class, input.ID,
-		additional.Properties{}, nil, tenant)
-	if err != nil {
-		errnf := ErrNotFound{}
-		if errors.As(err, &errnf) {
-			return &Error{"source object", StatusNotFound, err}
-		} else if errors.As(err, &ErrMultiTenancy{}) {
-			return &Error{"source object", StatusUnprocessableEntity, err}
-		}
-
-		return &Error{"source object", StatusInternalServerError, err}
-	}
-	input.Class = res.ClassName
 
 	unlock, err := m.locks.LockSchema()
 	if err != nil {
 		return &Error{"cannot lock", StatusInternalServerError, err}
 	}
 	defer unlock()
 
-	class, schemaVersion, err := input.validate(ctx, principal, m.schemaManager)
-	if err != nil {
+	if input.validateSchema(class) != nil {
 		if deprecatedEndpoint { // for backward comp reasons
 			return &Error{"bad inputs deprecated", StatusNotFound, err}
 		}
```

Comment:
```suggestion
	if err := input.validateSchema(class); err != nil {
		if deprecatedEndpoint { // for backward comp reasons
			return &Error{"bad inputs deprecated", StatusNotFound, err}
		}
```

I might be missing something, but wouldn't this Error have [the `err` from above](https://github.com/weaviate/weaviate/pull/6991/files#diff-1b8c2aafa295c0576e5b2bfd90244fe9e2af710e73fc6cbdc4493c785175eb19R98) if we do not assign it in this block?

Reply:
ah yes, nice catch!